prowler/providers/aws/services/iam/lib/privilege_escalation.py

```diff
@@ -0,0 +1,218 @@
+
+                    # len() == 1, so *
+                    elif len(api_action) == 1:
+                        api = api_action[0]
+                        # Add permissions if the API is present
+                        if api == "*":
+                            policies_combination.add(val)
+
+    return policies_combination
+
+
+def check_privilege_escalation(policy: dict) -> str:
+    """
+    check_privilege_escalation checks if the policy allows privilege escalation.
+    Args:
+        policy (dict): The policy to check.
+    Returns:
+        str: The policies affected by privilege escalation, separated by spaces.
+    """
+
+    policies_affected = ""
+
+    if policy:
+        allowed_actions = set()
+        denied_actions = set()
+        denied_not_actions = set()
+
+        statements = policy.get("Statement", [])
+        if not isinstance(statements, list):
+            statements = [statements]
+
+        for statement in statements:
+            effect = statement.get("Effect")
+            actions = statement.get("Action")
+            not_actions = statement.get("NotAction")
+
+            if effect == "Allow" and actions:
+                if isinstance(actions, str):
+                    allowed_actions.add(actions)
+                elif isinstance(actions, list):
+                    allowed_actions.update(actions)
+
+            if effect == "Deny" and actions:
+                if isinstance(actions, str):
+                    denied_actions.add(actions)
+                elif isinstance(actions, list):
+                    denied_actions.update(actions)
+
+            if effect == "Allow" and not_actions:
+                if isinstance(not_actions, str):
+                    denied_not_actions.add(not_actions)
+                elif isinstance(not_actions, list):
+                    denied_not_actions.update(not_actions)
+
+            if effect == "Deny" and not_actions:
+                if isinstance(not_actions, str):
+                    denied_not_actions.add(not_actions)
+                elif isinstance(not_actions, list):
+                    denied_not_actions.update(not_actions)
+
+        policies_combination = find_escalation_combinations(
+            allowed_actions, denied_actions, denied_not_actions
+        )
+
+        # Check all policies combinations and see if matchs with some combo key
+        combos = set()
+        for (
+            key,
+            values,
+        ) in privilege_escalation_policies_combination.items():
+            intersection = policies_combination.intersection(values)
+            if intersection == values:
+                combos.add(key)
+
+        if combos:
+            policies_affected = (
+                ", ".join(
+                    str(privilege_escalation_policies_combination[key])
+                    for key in combos
+                )
+                .replace("{", "")
+                .replace("}", "")
```

Comment:
Why this?

Reply:
It is for improve the `status_extended` format because the policies affected are added at the end of it, here is an example with the replace and without:

With replace:
`"'iam:AttachGroupPolicy', 'iam:SetDefaultPolicyVersion', 'iam:CreateLoginProfile', 'iam:AttachRolePolicy', 'iam:AddUserToGroup', 'iam:Put*', 'iam:PassRole', 'iam:*', 'iam:UpdateLoginProfile', 'iam:PutUserPolicy', 'iam:CreatePolicyVersion', 'iam:PutRolePolicy', 'iam:UpdateAssumeRolePolicy', 'iam:CreateAccessKey', 'iam:PutGroupPolicy', 'iam:AttachUserPolicy'"`

Without replace:
`"{'iam:PassRole'}, {'iam:Put*'}, {'iam:AttachRolePolicy'}, {'iam:CreateLoginProfile'}, {'iam:UpdateAssumeRolePolicy'}, {'iam:*'}, {'iam:SetDefaultPolicyVersion'}, {'iam:AddUserToGroup'}, {'iam:CreateAccessKey'}, {'iam:CreatePolicyVersion'}, {'iam:AttachUserPolicy'}, {'iam:PutGroupPolicy'}, {'iam:PutUserPolicy'}, {'iam:AttachGroupPolicy'}, {'iam:PutRolePolicy'}, {'iam:UpdateLoginProfile'}"`